### PR TITLE
[Bug] Fix constant In Predicate result error

### LINKED ISSUE #3510: [Bug] Constant InPredicate result error
**Describe the bug**
`select 1 not in (2, NULL, 1);`

expected: `0`
Actual: `NULL`

### PR DESCRIPTION
**Pull request by morningman**
`select 1 not in (2, NULL, 1);` should return `0`

Fix: #3510

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/analysis/InPredicate.java [hunk 1]
```diff
@@ -54,6 +54,8 @@
     private static final String IN = "in";
     private static final String NOT_IN = "not_in";
 
+    private static final NullLiteral NULL_LITERAL = new NullLiteral();
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/analysis/InPredicate.java:57]: Why use static variable?  I think local variable is enough.

File: fe/src/test/java/org/apache/doris/planner/QueryPlanTest.java [hunk 2]
```diff
@@ -725,4 +725,58 @@ public void testJoinPredicateTransitivityWithSubqueryInWhereClause() throws Exce
         Assert.assertTrue(explainString.contains("CROSS JOIN"));
         Assert.assertTrue(!explainString.contains("PREDICATES"));
     }
+
+    @Test
+    public void testConstantInPredicate() throws Exception {
```
Comment (kangkaisen) [inline fe/src/test/java/org/apache/doris/planner/QueryPlanTest.java:730]: please move the test to `ConstantExpressTest`

Comment (morningman) [reply fe/src/test/java/org/apache/doris/planner/QueryPlanTest.java:730]: OK

File: fe/src/main/java/org/apache/doris/analysis/InPredicate.java (hunk 1 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/analysis/InPredicate.java:57]: This object does not need to have one for every `InPredicate` instance.

**Review by kangkaisen — APPROVED** [review]
+1, LGTM

**Pull request merged**